odo/backends/bcolz.py

```diff
@@ -82,6 +101,22 @@ def resource_bcolz(uri, dshape=None, expected_dshape=None, **kwargs):
         dshape = datashape.dshape(dshape)
 
         dt = dshape_to_numpy(dshape)
+        if isrecord(dshape.measure):
+            dtype_pairs = [(name, dt[name].type) for name in dt.names]
+            is_object = [dtype == np.object_ for _, dtype in dtype_pairs]
+            if any(is_object):
+                raise TypeError("object dtypes are not supported by bcolz.\n"
```

Comment:
Looks like the fix is on the way for this:

https://github.com/Blosc/bcolz/pull/242